nextjs-13/src/lib/handler/base.handler.ts

```diff
@@ -0,0 +1,37 @@
+import { User } from '@prisma/client';
+import { NextResponse, NextRequest } from 'next/server';
+import { getToken } from 'next-auth/jwt';
+
+import { findUserById } from 'repositories/user.repository';
+
+export default async function baseHandler(
+  req: NextRequest,
+  callback: (currentUser: User, body: unknown) => Promise<NextResponse>
+) {
+  try {
+    const token = await getToken({ req });
+
+    if (token && token.userId) {
+      const currentUser = await findUserById(token.userId as string);
+
+      if (currentUser) {
+        let body = {};
+
+        try {
+          body = await req.json();
+        } catch (err) {}
+
+        console.log('========>body : ', body);
+        return await callback(currentUser, body);
+      }
+    }
+
+    return NextResponse.json({ message: 'Invalid token' }, { status: 401 });
+  } catch (err) {
+    console.error('========>err : ', err);
+    return NextResponse.json(
+      { message: (err as Error).message },
+      { status: 500 }
```

Comment:
`500` errors should never be raised by the application on purpose. `500` is reserved for unhandled errors. Here we could raise a `400` error (bas request) 💡

Reply:
But I really want to use this code to handle the unhandled errors 🤔 If we use 400, and the errors is on our side, then I think it may cause confusion to the client. What do you think about this? Or should we use other `5xx` error status codes?

And yeah, seems like I forgot to use the `http-status-codes` here, let me update it :D

Reply:
[^ Done using http status codes instead ✅ ](https://github.com/nimblehq/js-framework-benchmark/pull/96/commits/455c612b291d7e0f3ddf58b4f2b24ae0f3e08756#diff-10b91e667909ffe6c8b2d34d735e0fa46be166ba37cf1b5ec24f7ea6b54658bbR27)